sycl/plugins/unified_runtime/ur/adapters/level_zero/kernel.cpp

```diff
@@ -55,12 +55,20 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
 
   // If there are any pending arguments set them now.
   for (auto &Arg : Kernel->PendingArguments) {
+    // if the argument has access_mode_t::unknown, then this is a
+    // regular pointer
+    if (Arg.AccessMode == ur_mem_handle_t_::access_mode_t::unknown) {
```

Comment:
relying on "unknown" access mode seems at least subtle, suggest making it more robust with std::variant